plugins/discourse-details/test/javascripts/integration/prosemirror-editor/details-test.js

```diff
@@ -0,0 +1,77 @@
+import { click } from "@ember/test-helpers";
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import {
+  setupRichEditor,
+  testMarkdown,
+} from "discourse/tests/helpers/rich-editor-helper";
+
+module(
+  "Integration | Component | prosemirror-editor - details extension",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    const testCases = {
+      details: [
+        [
+          `[details="Summary"]This text will be hidden[/details]`,
+          `<details><summary>Summary</summary><p>This text will be hidden</p></details>`,
+          `[details="Summary"]\nThis text will be hidden\n\n[/details]\n\n`,
+        ],
+      ],
+      "details with open attribute": [
+        [
+          `[details="Summary" open]This text will be hidden[/details]`,
+          `<details open="true"><summary>Summary</summary><p>This text will be hidden</p></details>`,
+          `[details="Summary" open]\nThis text will be hidden\n\n[/details]\n\n`,
+        ],
+      ],
+      "details without summary": [
+        [
+          `[details]This text will be hidden[/details]`,
+          `<details><summary></summary><p>This text will be hidden</p></details>`,
+          `[details]\nThis text will be hidden\n\n[/details]\n\n`,
+        ],
+      ],
+      "details without summary but with open attribute": [
+        [
+          `[details open]This text will be hidden[/details]`,
+          `<details open="true"><summary></summary><p>This text will be hidden</p></details>`,
+          `[details open]\nThis text will be hidden\n\n[/details]\n\n`,
+        ],
+      ],
+    };
+
+    Object.entries(testCases).forEach(([name, tests]) => {
+      tests.forEach(([markdown, expectedHtml, expectedMarkdown]) => {
+        test(name, async function (assert) {
+          this.siteSettings.rich_editor = true;
+
+          await testMarkdown(assert, markdown, expectedHtml, expectedMarkdown);
+        });
+      });
+    });
+
+    test("opens and closes details on click", async function (assert) {
+      this.siteSettings.rich_editor = true;
+      const detailsMarkdown = `[details="Summary"]This text will be hidden[/details]`;
+      await setupRichEditor(assert, detailsMarkdown);
+
+      const detailsCss = ".d-editor-input details";
+      const summaryCss = `${detailsCss} summary`;
+      assert.dom(`${detailsCss} p`).isNotVisible();
+
+      await click(`${summaryCss}`);
+      assert.dom(`${detailsCss}`).hasAttribute("open");
+      assert.dom(`${detailsCss} p`).isVisible();
+
+      // NOTE (martin): For some reason this is just refusing to work, the
```

Comment:
This was driving me nuts, you're welcome to try and fix

Reply:
It was selecting the summary text with the double-ish click, it passes with a click elsewhere before clicking on the summary again.

Reply:
Ah thanks!